### PR TITLE
ISPN-8548 Clear post notifications can cause memory issues as it holds

### PR DESCRIPTION
**Pull request by wburns**
all entries locally

https://issues.jboss.org/browse/ISPN-8548

### CODE REVIEW

**Comment by wburns** [comment]
This should be cherry-picked into 9.1.x as well.

File: core/src/main/java/org/infinispan/interceptors/locking/ClusteringDependentLogic.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.Iterator;
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/interceptors/locking/ClusteringDependentLogic.java]: Unused import, checkstyle error.

Comment (wburns) [reply core/src/main/java/org/infinispan/interceptors/locking/ClusteringDependentLogic.java]: Fixed

**Comment by danberindei** [comment]
Integrated, thanks Will!

**Comment by gustavonalle** [comment]
Did CI actually run complete for this one? I can see only two attempts in vain @ http://ci.infinispan.org/job/Infinispan/job/PR-5596/

**Pull request merged**